rofl-scheduler/src/manager.rs

```diff
@@ -0,0 +1,1334 @@
+                "offer" => ?instance.offer,
+            );
+
+            // Instance seems acceptable.
+            local_state.accept.push(instance.id);
+            local_state.accepted.insert(instance.id, instance.clone());
+            local_state.resources_used = new_resource_use;
+            if let Some(deployment) = instance.deployment.clone() {
+                local_state
+                    .pending_start
+                    .push((instance, deployment.clone(), true));
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Process queued jobs.
+    async fn process_jobs(self: &Arc<Self>, local_state: &mut LocalState) -> Result<()> {
+        // Prepare job to accept instances.
+        let accept_jobs: Vec<_> = local_state
+            .accept
+            .chunks(16)
+            .map(|ids| self.clone().accept_instances(ids.to_vec()))
+            .collect();
+
+        // Prepare jobs to remove instances.
+        let remove_jobs: Vec<_> = local_state
+            .maybe_remove
+            .iter()
+            .map(|(id, ts)| self.clone().maybe_remove_instance(*id, *ts))
+            .collect();
+
+        // Prepare jobs to start instances.
+        let start_jobs: Vec<_> = local_state
+            .pending_start
+            .iter()
+            .map(|(instance, deployment, wipe_storage)| {
+                self.clone()
+                    .start_instance(instance.clone(), deployment.clone(), *wipe_storage)
+            })
+            .collect();
+
+        // Prepare jobs to stop instances.
+        let stop_jobs: Vec<_> = local_state
+            .pending_stop
+            .iter()
+            .map(|(id, wipe_storage)| self.clone().stop_instance(*id, *wipe_storage))
+            .collect();
+
+        // Prepare jobs to claim payments.
+        let claim_payment_jobs: Vec<_> = local_state
+            .claim_payment
+            .chunks(16)
+            .map(|chunk| self.clone().claim_payment(chunk.to_vec()))
+            .collect();
+
+        // Execute all jobs in parallel.
+        let mut jobs = Vec::new();
+        for job in accept_jobs {
+            jobs.push(tokio::spawn(job));
+        }
+        for job in remove_jobs {
+            jobs.push(tokio::spawn(job));
+        }
+        for job in start_jobs {
+            jobs.push(tokio::spawn(job));
+        }
+        for job in stop_jobs {
+            jobs.push(tokio::spawn(job));
+        }
+        for job in claim_payment_jobs {
+            jobs.push(tokio::spawn(job));
+        }
+
+        slog::info!(self.logger, "running jobs"; "num_jobs" => jobs.len());
+
+        for job in jobs {
+            match job.await {
+                Err(err) => {
+                    slog::error!(self.logger, "task panicked"; "err" => ?err);
+                }
+                Ok(Err(err)) => {
+                    slog::error!(self.logger, "task failed"; "err" => ?err);
+                }
+                Ok(Ok(_)) => {
+                    // Ok.
```

Comment:
Should we add some debug logging here for each finished task? Might be helpful for debugging in case there would be issues with specific tasks executing/blocking for long time?